### PR TITLE
Convert to csv script generalization

### PR DESCRIPTION
**Pull request by Yueqiao12Zhang**
Move the input entry into the command line with the specification of the input file's entity type. This makes the script more general in the way that the user does not have to go inside the script to change the output entity type, which makes conversion from CSV to RDF much simpler. (If the output entity type is not specified, then none of the triple in the RDF will have type, and the key attribute will be difficult to distinguish.)
The two test files were small examples, they were used to test the script at development and they were not enough to test the script after it was almost finished. The new test files will be added in later PRs.

### CODE REVIEW

File: musicbrainz/csv/convert_to_csv.py
Comment (dchiller) [inline musicbrainz/csv/convert_to_csv.py:1]: Like in the #58, can you add a module docstring that explains about that command-line arguments?

**Comment by dchiller** [comment]
> Also adds the .history directory (used by the Local History VSCode extension) to .gitignore

You can remove this...because this PR does not add `.history` directory.

**Comment by dchiller** [comment]
> Extract the Wikidata automatically.

What does this mean...I don't see anything in this PR about extracting anything from Wikidata?

**Comment by dchiller** [comment]
> Concatenate the IDs with URLs with specific types automatically.

I'm not sure what this means...

**Comment by dchiller** [comment]
> It also extracts the Wikidata reference from the `relations > resource > url` object if MusicBrainz reconciled this entity already.

What line in the code introduced in this PR does this?

**Comment by dchiller** [comment]
> If the object contains "_id", it can refer to a MusicBrainz page in the way of string concatenation. The script changes the id to `https://musicbrainz.org/{the_type_of_the_id}/{id}`.

What line of code introduced in this PR does this?

File: musicbrainz/csv/convert_to_csv.py [hunk 1]
```diff
@@ -1,29 +1,38 @@
...
-# https://data.metabrainz.org/pub/musicbrainz/data/json-dumps/
-# Output file can be imported to OpenRefine
-# Both input and output files must be in data folder
-# Second input is the type of entities in the CSV file.
-# Example command: python3 convert_to_csv.py test2 recording
-# The input filename can be directly entered in second argument of the commandline
+"""
+Input file can be downloaded in
+https://data.metabrainz.org/pub/musicbrainz/data/json-dumps/
+The input files should be in data folder, the output file will be generate in the data folder
```
[the remark this answers is not in the document]
Comment (Yueqiao12Zhang) [reply musicbrainz/csv/convert_to_csv.py:4]: It's the ./data folder in the musicbrainz/csv directory.

**Comment by Yueqiao12Zhang** [comment]
> > It also extracts the Wikidata reference from the `relations > resource > url` object if MusicBrainz reconciled this entity already.
> 
> What line in the code introduced in this PR does this?

101-108,
```
    elif isinstance(data, list):
        # extract each element of the list.
        if key == "relations":
            for element in data:
                if "type" in element and element["type"] == "wikidata":
                    extract(
                        (element["url"])["resource"], value, first_level, key + "_wiki"
                    )
```

**Comment by Yueqiao12Zhang** [comment]
> > If the object contains "_id", it can refer to a MusicBrainz page in the way of string concatenation. The script changes the id to `https://musicbrainz.org/{the_type_of_the_id}/{id}`.
> 
> What line of code introduced in this PR does this?

80-91,
```
            for k in data:
                if k == "id":
                    # extract its id
                    keywords = key.split("_")
                    word = keywords[-1]
                    id = data["id"]
                    extract(
                        f"https://musicbrainz.org/{word}/{id}",
                        value,
                        first_level,
                        key + "_id",
                    )

```

File: musicbrainz/csv/convert_to_csv.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dchiller) [reply musicbrainz/csv/convert_to_csv.py:4]: I think this contradicts line 31. It also contradicts the instructions in line 8.

**Comment by dchiller** [comment]
None of the lines of code you mention are added or changed in this PR!

**Comment by Yueqiao12Zhang** [comment]
Oh, it was changed before this PR, I got confused during the rebase process. I will change the PR description.

File: musicbrainz/csv/convert_to_csv.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dchiller) [reply musicbrainz/csv/convert_to_csv.py:4]: @Yueqiao12Zhang

**Pull request merged**